File: FrontTP8/src/types/Pedido.ts
import PedidoDetalle from "./PedidoDetalle";

class Pedido {
    fechaPedido: Date = new Date();
    totalPedido: number = 0;
    detalles: Array<PedidoDetalle> = new Array<PedidoDetalle>();
}

export default Pedido;

File: BackTP8/src/main/java/com/example/backtp4/model/Pedido.java
package com.example.backtp4.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Pedido {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date fechaPedido;
    private Double totalPedido;
    @OneToMany(mappedBy = "pedido", cascade = CascadeType.ALL)
    private List<PedidoDetalle> detalles = new ArrayList<>();
}
